### PR TITLE
Goodbye Bootstrap

### PR DESCRIPTION
**Pull request by alexpaxton**
- [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### The Problem
- Bootstrap is bulky
- Bootstrap theme is bulky by proxy
- jQuery is in the project only because Bootstrap uses it, be we aren't using the parts of bootstrap that jQuery powers
- It's been a while since we've inventoried which parts of the theme we are actually using

### The Solution
- Remove Bootstrap & jQuery
- Bring in stylesheets from theme and cleanup formatting
- Recreate the 12 column grid

### Notes
- This removes a massive amount of CSS from the project with almost no noticeable difference
- We were using surprisingly much less of the theme than I thought
- The theme is fully exposed inside Chronograf now
- Opens up a lot of opportunity to organize and cleanup styles further
- A lot of components had excessive markup because of their roots in bootstrap, these can be simplified
- Will be useful in the future when Chronograf exposes its graphing library and it won't need Bootstrap or jQuery

### CODE REVIEW

File: ui/src/kapacitor/components/KapacitorRules.js [hunk 1]
```diff
@@ -54,51 +54,50 @@ const KapacitorRules = ({
...
-        <h2 className="panel-title">
-          {builderHeader}
-        </h2>
-        <div className="u-flex u-ai-center u-jc-space-between">
+      <div className="panel">
+        <div className="panel-heading">
+          <h2 className="panel-title">
+            {builderHeader}
+          </h2>
+          <div className="u-flex u-ai-center u-jc-space-between">
```
Comment (lukevmorris) [inline ui/src/kapacitor/components/KapacitorRules.js]: do we need these classes?

File: ui/src/style/chronograf.scss [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 // Fonts
 @import 'fonts/fonts';
+@import 'fonts/icon-font';
 
-// Base Theme
-@import 'theme/bootstrap';
-@import 'theme/bootstrap-theme';
-@import 'theme/reset';
+// Theme
+@import 'theme/chronograf-theme';
```
Comment (lukevmorris) [inline ui/src/style/chronograf.scss]: do we need some sort of reset to replace what came with `theme/reset`?

File: ui/src/style/components/graph.scss [hunk 3]
```diff
@@ -76,15 +70,20 @@ $graph-gutter: 16px;
...
+}
+.graph-heading > input[type="text"].form-control,
+.graph-name {
+  height: 30px;
   font-weight: 600;
-  @include no-user-select();
-  transition:
-    color 0.25s ease;
+  padding: 0 11px;
+  letter-spacing: 0px;
```
Comment (lukevmorris) [inline ui/src/style/components/graph.scss]: i think this can be just 0

File: ui/src/style/components/search-widget.scss [hunk 4]
```diff
@@ -1,41 +1,37 @@
...
-.users__search-widget {
+.search-widget {
   position: relative;
 
+  &:first-child {
+    margin-right: 8px;
+  }
+
+  &:only-child {
+    margin-right: none;
```
Comment (lukevmorris) [inline ui/src/style/components/search-widget.scss]: I'm not sure `none` is valid here?

File: ui/src/style/fonts/icon-font.scss [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  &.google3:before {content: "\ea8a";}
+  &.google-plus:before {content: "\ea8b";}
+  &.facebook:before {content: "\ea90";}
+  &.facebook2:before {content: "\ea91";}
+  &.twitter:before {content: "\ea96";}
+  &.github:before {content: "\eab0";}
+  &.trello:before {content: "\eab3";}
+  &.linkedin:before {content: "\eac9";}
+  &.linkedin2:before {content: "\eaca";}
+  &.stackoverflow:before {content: "\ead0";}
```
Comment (lukevmorris) [inline ui/src/style/fonts/icon-font.scss]: Do we need these 3rd party brand icons?

File: ui/src/style/modules/mixins.scss [hunk 6]
```diff
@@ -115,7 +115,7 @@ $scrollbar-offset: 3px;
     cursor: default;
   }
 }
-.no-user-select {
+%no-user-select {
```
Comment (lukevmorris) [inline ui/src/style/modules/mixins.scss]: ahhh nice, perfect usecase

File: ui/src/style/theme/_buttons.scss [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+@mixin btn-outline-styles (
+    $text-color,
+    $text-color-hover,
+    $border-color,
+    $border-color-hover,
+    $border-color-press
+  ) {
+
+  // Adjusting line height in response to more borders
+  line-height: ($form-md-height - 4px);
```
Comment (lukevmorris) [inline ui/src/style/theme/_buttons.scss]: Do we need calcs here?

File: ui/src/style/theme/_tables.scss [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  border-collapse: collapse;
+  margin: 0;
+
+  th, td {
+    font-size: 13px;
+    text-align: left;
+
+    &.monotype {
+      font-family: $code-font;
+      letter-spacing: 0px;
```
Comment (lukevmorris) [inline ui/src/style/theme/_tables.scss]: 0!

File: ui/src/style/theme/_tables.scss [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    border-bottom: $ix-border solid $table--border-color;
+  }
+
+  // Body
+  > tbody > tr > td, {
+    text-align: left;
+    color: $g13-mist;
+    font-weight: 500;
+    padding: 4px 8px;
+    border: 0px;
```
Comment (lukevmorris) [inline ui/src/style/theme/_tables.scss]: 0!

File: ui/src/kapacitor/components/KapacitorRules.js (hunk 1 above)
Comment (lukevmorris) [reply ui/src/kapacitor/components/KapacitorRules.js]: They were removed in a bunch of other places, wondering if they should be removed here

File: ui/src/style/theme/_buttons.scss (hunk 7 above)
Comment (alexpaxton) [reply ui/src/style/theme/_buttons.scss]: Might not need them, had them in there to be cautious. I think the compiler should know to evaluate the expression without the explicit `()` around it

File: ui/src/kapacitor/components/KapacitorRules.js (hunk 1 above)
Comment (alexpaxton) [reply ui/src/kapacitor/components/KapacitorRules.js]: ah good catch, I don't think that div is actually doing anything

File: ui/src/style/fonts/icon-font.scss (hunk 5 above)
Comment (alexpaxton) [reply ui/src/style/fonts/icon-font.scss]: Arguably no. They're vestigial from when the theme was used on the website

**Pull request merged**